doc-processor-gradle-plugin/src/main/kotlin/nl/jolanrensen/docProcessor/gradle/ProcessDocTask.kt

```diff
@@ -0,0 +1,187 @@
+
+    /**
+     * Where the generated sources are placed.
+     */
+    @get:OutputFiles
+    val targets: FileCollection = factory.fileCollection()
+
+    /**
+     * Whether to print debug information.
+     */
+    @get:Input
+    val debug: Property<Boolean> = factory
+        .property(Boolean::class.java)
+        .convention(false)
+
+    /**
+     * The limit for while-true loops in processors. This is to prevent infinite loops.
+     */
+    @get:Input
+    val processLimit: Property<Int> = factory
+        .property(Int::class.java)
+        .convention(10_000)
+
+    /**
+     * The processors to use. These must be fully qualified names, such as:
+     * `"com.example.plugin.MyProcessor"`
+     */
+    @get:Input
+    val processors: ListProperty<String> = factory
+        .listProperty(String::class.java)
+        .convention(emptyList())
+
+    /** The classpath of this task. */
+    @Classpath
+    val classpath: Configuration = project.maybeCreateRuntimeConfiguration()
+
+    /** Used by the task to execute [ProcessDocsGradleAction]. */
+    @get:Inject
+    abstract val workerExecutor: WorkerExecutor
+
+    private fun Project.maybeCreateRuntimeConfiguration(): Configuration =
+        project.configurations.maybeCreate("kotlinKdocIncludePluginRuntime") {
+            isCanBeConsumed = true
+            dependencies.add(project.dependencies.create("org.jetbrains.kotlin:kotlin-compiler:1.7.20"))
+            dependencies.add(project.dependencies.create("org.jetbrains.dokka:dokka-analysis:1.7.20")) // compileOnly in base plugin
+            dependencies.add(project.dependencies.create("org.jetbrains.dokka:dokka-base:1.7.20"))
+            dependencies.add(project.dependencies.create("org.jetbrains.dokka:dokka-core:1.7.20"))
+        }
+
+    private fun <T : Any> NamedDomainObjectContainer<T>.maybeCreate(name: String, configuration: T.() -> Unit): T =
+        findByName(name) ?: create(name, configuration)
+
+    /**
+     * Adds dependency to plugin.
+     * Don't forget to add any new processor to the [processors] list.
+     *
+     * @param dependencyNotation Dependency notation
+     *
+     * @see org.gradle.api.artifacts.dsl.DependencyHandler.create
+     */
+    fun addPlugins(vararg dependencyNotation: Any) {
+        classpath.dependencies.addAll(
+            dependencyNotation.map { project.dependencies.create(it) }
+        )
+    }
+
+    /**
+     * Adds dependency to plugin.
+     * Don't forget to add any new processor to the [processors] list.
+     *
+     * @param dependency Dependency
+     */
+    fun addPlugins(vararg dependency: Dependency) {
+        classpath.dependencies.addAll(dependency)
+    }
+
+    private fun println(message: Any?) {
+        if (debug.get()) {
+            kotlin.io.println(message)
+        }
+    }
+
+    init {
+        outputs.upToDateWhen { false }
+    }
+
+    @TaskAction
+    fun process() {
+        println("Hello from plugin 'nl.jolanrensen.docProcessor'")
```

Comment:
Ahaha:) If you will move it under the JetBrains, I suggest to rename it to the 'Hello from JetBrains"!